server/src/main/java/org/opensearch/cluster/action/index/NodeMappingRefreshAction.java

```diff
@@ -80,11 +92,70 @@ public NodeMappingRefreshAction(TransportService transportService, MetadataMappi
     }
 
     public void nodeMappingRefresh(final DiscoveryNode clusterManagerNode, final NodeMappingRefreshRequest request) {
+        new NodeMappingRefreshClusterManagerAction(clusterManagerNode, request).run();
+    }
+
+    private void sendNodeMappingRefreshToClusterManager(
+        final DiscoveryNode clusterManagerNode,
+        final NodeMappingRefreshRequest request,
+        ActionListener listener
+    ) {
         if (clusterManagerNode == null) {
             logger.warn("can't send mapping refresh for [{}], no cluster-manager known.", request.index());
             return;
         }
-        transportService.sendRequest(clusterManagerNode, ACTION_NAME, request, EmptyTransportResponseHandler.INSTANCE_SAME);
+        transportService.sendRequest(clusterManagerNode, ACTION_NAME, request, new EmptyTransportResponseHandler(ThreadPool.Names.SAME) {
+            @Override
+            public void handleException(TransportException exp) {
+                listener.onFailure(exp);
+            }
+        });
+    }
+
+    /**
+     * RetryableAction for performing retires for cluster manager throttling.
+     */
```

Comment:
can we extend `TransportClusterManagerNodeAction` for same instead of writing another `Action` ?

Reply:
Actually I tried to check that as well, for that we might need more changes. Will give it one more look.

Anyway I am going to remove this changes from here, will add it back when we onboard this task type to framework. We can discuss more in that PR.